dataactcore/interfaces/function_bag.py

```diff
@@ -296,7 +296,7 @@ def mark_job_status(job_id, status_name):
 
     # if status is changed to finished for the first time, check dependencies
     # and add to the job queue as necessary
-    if old_status != 'finished' and status_name == 'finished':
+    if old_status != 'finished' and status_name == 'finished' and not skip_check:
```

Comment:
They get updated to `finished` here if there are no results: https://github.com/fedspendingtransparency/data-act-broker-backend/pull/689/files#diff-7507a7c0c5557b3a9aabb9c8c18c4e5bR494

And here if there is actual data: https://github.com/fedspendingtransparency/data-act-broker-backend/blob/7e8c5df31f4c4a27a5e2785b27cf403b5ce797b4/dataactbroker/handlers/fileHandler.py#L572